docs/enhanced-provider-k8s.md

```diff
@@ -0,0 +1,303 @@
+
+Apparently, the solution to this problem is turning unmanaged external resource into managed one. In our case, this can be done by defining an `Object` resource to represent the `ClusterServiceVersion` resource. Then, the deletion of `Object` resource will trigger the deletion of `ClusterServiceVersion` resource. There is no need to define `Object` for `InstallPlan` because when `ClusterServiceVersion` is deleted, it will trigger the deletion of `InstallPlan` automatically which is handled by OLM.
+
+One interesting thing for the `Object` resource to be defined is that, the provider only needs to take care of the deletion and observation for the `ClusterServiceVersion` resource. It should not be responsible for the creation and update because these operations are handled by OLM. This scenario leads to the introduction of the feature **Resource Management Policy** in Provider Kubernetes.
+
+### Resource Management Policy
+
+The resource management policy is used to instruct the provider to manage Kubernetes resources in a fine-grained way. Given that:
+
+* To manage an external resource that can not be observed is usually meaningless.
+* Create and update are usually coupled together.
+
+We only need 4 types of management policies:
+
+| Observe | Create & Update | Delete | Management Policy      | Memo
+|:--------|:----------------|:-------|:-----------------------|:----
+| Y       | Y               | Y      | Default                | The provider can fully manage the resource. This is the default policy.
+| Y       | Y               | N      | Undeletable            | The provider can observe, create, or update the resource, but can not delete it. 
+| Y       | N               | Y      | ObservableAndDeletable | The provider can observe or delete the resource, but can not create and update it.
+| Y       | N               | N      | Observable             | The provider can only observe the resource. This maps to the read-only scenario where the resource is fully controlled by third party application and the provider only reads the status of the resource.
+
+The idea of Resource Management Policy is actually expanded from the concept of `DeletionPolicy` in Crossplane which can be found in Crossplane code:
+
+```go
+const (
+    // DeletionOrphan means the external resource will orphaned when its managed
+    // resource is deleted.
+    DeletionOrphan DeletionPolicy = "Orphan"
+
+    // DeletionDelete means both the external resource will be deleted when its
+    // managed resource is deleted.
+    DeletionDelete DeletionPolicy = "Delete"
+)
+```
+
+The `DeletionOrphan` policy is actually equivalent to the `Undeletable` resource management policy.
```

Comment:
Agreed. Ideally this could be part of Crossplane core. Maybe expand the current `DeletionPolicy` in Crossplane runtime, rename `DeletionPolicy` to be something more generic such as `ManagementPolicy`, and `Undeletable` is equivalent to `DeleteOrphen`.

Of course, in that case, we need to consider backward compatibility seriously since it requires CRD change. That is why I use annotation instead of modifying CRD to implement this feature at the moment, since that makes it easier in future if we decide to uplift it to Crossplane core per community usage feedback.

Reply:
Yeah, I see and agree with that. But it would be helpful to **explicitly** mention how to handle any conflict/contention until this is bubbles up to the runtime.